lib/edit-site-page.php

```diff
@@ -144,30 +144,11 @@ function gutenberg_edit_site_init( $hook ) {
 	}
 	$settings['styles'] = gutenberg_get_editor_styles();
 
-	$template_ids      = array();
-	$template_part_ids = array();
-	foreach ( get_template_types() as $template_type ) {
-		// Skip 'embed' for now because it is not a regular template type.
-		// Skip 'index' because it's a fallback that we handle differently.
-		if ( in_array( $template_type, array( 'embed', 'index' ), true ) ) {
-			continue;
-		}
-
-		$current_template = gutenberg_find_template_post_and_parts( $template_type );
-		if ( isset( $current_template ) ) {
-			$template_ids[ $template_type ] = $current_template['template_post']->ID;
-			$template_part_ids              = $template_part_ids + $current_template['template_part_ids'];
-		}
-	}
-
 	$settings['editSiteInitialState'] = array();
 
-	$settings['editSiteInitialState']['templateType']    = 'wp_template';
-	$settings['editSiteInitialState']['templateIds']     = array_values( $template_ids );
-	$settings['editSiteInitialState']['templatePartIds'] = array_values( $template_part_ids );
-
-	$settings['editSiteInitialState']['showOnFront'] = get_option( 'show_on_front' );
-	$settings['editSiteInitialState']['page']        = array(
+	$settings['editSiteInitialState']['templateType'] = 'wp_template';
+	$settings['editSiteInitialState']['showOnFront']  = get_option( 'show_on_front' );
+	$settings['editSiteInitialState']['page']         = array(
```

Comment:
This could then be computed in the client.

Reply:
Sounds good. I'd rather tackle it in a follow-up though, to keep things bite-sized.

Reply:
One thought for the follow up task is that it would be nice to have this be extensible in some way. (e.g. a way to say what initial page is loaded) You could accomplish a similar thing with `setPage` client side, but it would be nice to avoid doing that fetch twice.